src/System.Management.Automation/engine/remoting/fanin/WSManTransportManager.cs

```diff
@@ -1546,8 +1546,14 @@ private void Initialize(Uri connectionUri, WSManConnectionInfo connectionInfo)
                 throw new PSRemotingTransportException(PSRemotingErrorId.ConnectFailed, RemotingErrorIdStrings.BasicAuthOverHttpNotSupported);
             }
 
-            // Allow HTTPS on Unix only if SkipCACheck and SkipCNCheck are selected, because OMI client does not support validating server certificates.
-            if (isSSLSpecified && (!connectionInfo.SkipCACheck || !connectionInfo.SkipCNCheck))
+            // The OMI client distributed with PowerShell does not support validating server certificates. Check if
+            // psrpclient supports the CA check to determine if a custom build of psrpclient and mi are present which
+            // do support verification. If WSManGetSessionOptionAsDword does not return 0 then it's not supported.
+            int sslResult;
+            bool verificationAvailable = WSManNativeApi.WSManGetSessionOptionAsDword(_wsManSessionHandle,
```

Comment:
Since `int sslResult` is not used we can remove it and do this:

```suggestion
            bool verificationAvailable = WSManNativeApi.WSManGetSessionOptionAsDword(_wsManSessionHandle, WSManNativeApi.WSManSessionOption.WSMAN_OPTION_SKIP_CA_CHECK, out _) == 0;
```

Reply:
Sure, I'm not the best in newer C# so I didn't think this was possible.

Reply:
Thanks for the suggestion, has been updated.